oauthenticator/github.py

```diff
@@ -109,20 +155,41 @@ def authenticate(self, handler, data=None):
         resp = yield http_client.fetch(req)
         resp_json = json.loads(resp.body.decode('utf8', 'replace'))
 
-        username =  resp_json["login"]
+        username = resp_json["login"]
 
         # Check if user is a member of any whitelisted organizations.
         # This check is performed here, as it requires `access_token`.
         if self.github_organization_whitelist:
             for org in self.github_organization_whitelist:
                 user_in_org = yield self._check_organization_whitelist(org, username, access_token)
-                if user_in_org:
-                    return username
-            else:  # User not found in member list for any organisation
-                return None
-        else:  # no organization whitelisting
-            return username
-
+                if not user_in_org:
+                    # User not found in member list for any organisation
+                    return None
```

Comment:
This has broken the logic of group whitelist checking. If the user is not in the first organization in the whitelist, they will not be allowed to login anymore. Let's try to leave the whitelist-checking logic unchanged.